AmplifyPlugins/API/Sources/AWSAPIPlugin/Operation/AWSRESTOperation.swift

```diff
@@ -126,6 +101,70 @@ final public class AWSRESTOperation: AmplifyOperation<
             let task = session.dataTaskBehavior(with: finalRequest)
             mapper.addPair(operation: self, task: task)
             task.resume()
+        case .failure(let error):
+            Amplify.API.log.debug("Dispatching error \(error)")
+            dispatch(result: .failure(error))
+            finish()
+        }
+    }
+
+    private func validateRequest(_ request: RESTOperationRequest) -> Result<RESTOperationRequest, APIError> {
+        do {
+            try request.validate()
+            return .success(request)
+        } catch let error as APIError {
+            return .failure(error)
+        } catch {
+            return .failure(APIError.unknown("Could not validate request", "", nil))
+        }
+    }
+
+    private func getEndpointConfig(
+        from request: RESTOperationRequest
+    ) -> Result<AWSAPICategoryPluginConfiguration.EndpointConfig, APIError> {
+        do {
+            return .success(try pluginConfig.endpoints.getConfig(for: request.apiName, endpointType: .rest))
+        } catch let error as APIError {
+            return .failure(error)
+        } catch {
+            return .failure(APIError.unknown("Could not get endpoint configuration", "", nil))
+        }
+    }
+
+    private func buildURLRequest(from request: RESTOperationRequest) -> Result<URLRequest, APIError> {
+        getEndpointConfig(from: request).flatMap { endpointConfig in
+            do {
+                let url = try RESTOperationRequestUtils.constructURL(
+                    for: endpointConfig.baseURL,
+                    withPath: request.path,
+                    withParams: request.queryParameters
+                )
+                return .success(RESTOperationRequestUtils.constructURLRequest(
+                    with: url,
+                    operationType: request.operationType,
+                    requestPayload: request.body
+                ))
+            } catch let error as APIError {
+                return .failure(error)
+            } catch {
+                return .failure(APIError.operationError("Failed to construct URL", "", error))
+            }
+        }
+    }
+
+    private func applyInterceptor(_ interceptor: URLRequestInterceptor, request: URLRequest) async -> Result<URLRequest, APIError> {
+        do {
+            return .success(try await interceptor.intercept(request))
+        } catch let error as APIError {
+            return .failure(error)
+        } catch {
+            return .failure(
+                APIError.operationError(
+                    "Failed to intercept request fully.",
+                    "Something wrong with the interceptor",
```

Comment:
can we provide more info from the error description and recovery message? such as `type(of: interceptor)` or `error.errorDescription` in the error description
```suggestion
                    "Failed to intercept request with \(type(of: interceptor)). Error message: \(error.errorDescription).",
                    "See underlying error for more details",
```

Reply:
sure, I was just copying the previous code error messages. We definitly can add more info here.